[unit][1]:

```
2019/04/05 21:17:38 Executing test unit
make: *** No rule to make target `check'.  Stop.
```

But you [have a `check` rule][1].  And for [images][2]:

```
2019/04/05 21:11:41 Build kube-etcd-signer-server failed, printing logs:
2019/04/05 21:11:41 Build kube-client-agent failed, printing logs:
Unable to retrieve container logs for docker://f106aa02ddee7d66fb5ea74e8c15bb42dc62f4ab4dfe430dbfccf97a10d85a1aUnable to retrieve container logs for docker://b73e6a230154ab7a5090719e7434b91ebe8425bec74ffd277198120784b6e25dUnable to retrieve container logs for docker://88f9c1ce71413b47b1c9fcb57ff57a3f66ca23700c3c691124189467580cef25Unable to retrieve container logs for docker://e035c30a5a1b53411563994db85bf93d97d3d71769d1fd406c5792d5c5d48e572019/04/05 21:11:54 Ran for 13s
error: could not run steps: some steps failed:
  * could not wait for build: the build kube-etcd-signer-server failed with reason GenericBuildFailed: Generic Build failure - check logs for details.
  * could not wait for build: the build kube-client-agent failed with reason GenericBuildFailed: Generic Build failure - check logs for details.
```

I suspect the CI cluster is just flaking on you, maybe `git commit --amend --no-edit` to get a fresh commit timestamp and force-push to get a fresh CI-cluster namespace?

[1]: https://github.com/openshift/kubecsr/blob/e94c059bd183986f6d0c9e7bb6f0902e67f11e38/Makefile#L11
[2]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_kubecsr/8/pull-ci-openshift-kubecsr-openshift-4.0-images/6/build-log.txt

--- a/pkg/certagent/agent.go
+++ b/pkg/certagent/agent.go
@@ -99,7 +99,8 @@ func GenerateCSRObject(config CSRConfig) (*capi.CertificateSigningRequest, error
 
 // RequestCertificate will create a certificate signing request for a node
 // with the config given and send it to a signer via a POST request.
-// If something goes wrong it returns an error.
+// If something goes wrong it returns an error but wait forever for
+// server to respond to request.
 // NOTE: This method does not return the approved CSR from the signer.
 func (c *CertAgent) RequestCertificate() error {
 	csr, err := GenerateCSRObject(c.config)
@@ -107,10 +108,16 @@ func (c *CertAgent) RequestCertificate() error {
 		return fmt.Errorf("error generating CSR Object: %v", err)
 	}
 
-	// send CSR to the signer
-	if _, err := c.client.Create(csr); err != nil {
-		return fmt.Errorf("error sending CSR to signer: %v", err)
-	}
+	duration := 10 * time.Second
+	// wait forever for success and retry every duration interval
+	wait.PollInfinite(duration, func() (bool, error) {
+		_, err := c.client.Create(csr)
+		if err != nil {
+			glog.Errorf("error sending CSR to signer: %v", err)
+			return false, nil
+		}
+		return true, nil
+	})
 
 	rcvdCSR, err := c.WaitForCertificate()
 	if err != nil {

